src/backend/distributed/executor/local_executor.c

```diff
@@ -60,13 +60,17 @@
  *		Note that for read-only queries, after the local execution, there is no
  *      need to kick in adaptive executor.
  *
+ * (4) Execution of multi shards local queries and
+ *     remote multi-shard queries within a transaction block
+ *
+ *      We prefer local execution when we are inside a transaction block, because not using
```

Comment:
minor: 
```
 because not using
local execution might create some visibility problems for other commands in the transaction
block.
```

I guess that was not the case, we used to error out on those cases. So, maybe reflect that in the comment such that it could lead to limitations if we don't do this?

Reply:
we would normally error out, but with the latest additional functionalities in local execution, to be able to enable local table join with reference table easily, we chose this way. IIRC, it would cause visibility problems in some commands. Normally this wasn't an issue because we didn't allow local table and reference table joins.

Does this make sense?

Reply:
the point is we would never enable any visibility issues, and error out in such cases. But, when  I read the comment, I feel like we're doing it otherwise we'd have visibility issues. Instead, we're doing it, otherwise we'd have to error out & have limitations

Reply:
I see your point, I will update the comment as that is the case now, but it wasn't the case before this PR: https://github.com/citusdata/citus/pull/3733/files we actually had a codepath that would use both a local execution and connections to localhost.

Reply:
> we actually had a codepath that would use both a local execution and connections to localhost.

I missed that, still it is a bug anyway, it shouldn't have happened :)